src/dataframe/dataframe.jl

```diff
@@ -1194,14 +1204,51 @@ function Base.append!(df1::DataFrame, df2::AbstractDataFrame)
     end
     ncol(df2) == 0 && return df1
 
-    _names(df1) == _names(df2) || throw(ArgumentError("Column names do not match"))
+    if cols == :orderequal && _names(df1) != _names(df2)
+        if ncol(df1) != ncol(df2)
+            ns1 = Set(_names(df1))
```

Comment:
No need to store values in temporary objects: just do `symdiff(_names(df1), _names(df2))`. Same in the other branch.

Reply:
fixed